From cd93006829d56782d605b4202c45e1aad7ce1746 Mon Sep 17 00:00:00 2001
From: Jehan <jehan@girinstud.io>
Date: Wed, 13 Aug 2014 05:08:08 +0000
Subject: [PATCH 1/3] Bug 731013 - cross-compilation broken when building
 extract-strings

It is actually a bad idea to use noinst_PROGRAMS for build tools,
because it adds a $(EXEEXT). It is best to override the all target
with all-local to trigger the tool build.
---
 gtk/Makefile.am           | 2 +-
 gtk/inspector/Makefile.am | 2 +-
 util/Makefile.am          | 6 +++---
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/gtk/Makefile.am b/gtk/Makefile.am
index 5c31084..7b78fb8 100644
--- a/gtk/Makefile.am
+++ b/gtk/Makefile.am
@@ -1566,7 +1566,7 @@ distclean-local:
 	if test $(srcdir) != .; then \
 	  rm -f $(MAINTAINERCLEANFILES); \
 	  rm -rf stock-icons; \
-	  rm -f extract-strings$(EXEEXT); \
+	  rm -f extract-strings$(BUILD_EXEEXT); \
 	  rm -f $(template_headers); \
 	fi
 
diff --git a/gtk/inspector/Makefile.am b/gtk/inspector/Makefile.am
index 9e14945..7b1876f 100644
--- a/gtk/inspector/Makefile.am
+++ b/gtk/inspector/Makefile.am
@@ -114,7 +114,7 @@ templates = 				\
 
 template_headers = $(templates:.ui=.ui.h)
 
-extract_strings = $(top_builddir)/util/extract-strings
+extract_strings = $(top_builddir)/util/extract-strings$(BUILD_EXEEXT)
 
 %.ui.h : %.ui 
 	$(AM_V_GEN) $(extract_strings) $< > $@
diff --git a/util/Makefile.am b/util/Makefile.am
index e8cf8fd..d0aa488 100644
--- a/util/Makefile.am
+++ b/util/Makefile.am
@@ -4,10 +4,10 @@ extract_strings_cppflags =
 extract_strings_cflags = $(GLIB_CFLAGS_FOR_BUILD)
 extract_strings_ldadd = $(GLIB_LIBS_FOR_BUILD)
 
-extract-strings$(EXEEXT): $(extract_strings_sources)
-	@rm -f extract-strings
+extract-strings$(BUILD_EXEEXT): $(extract_strings_sources)
+	@rm -f extract-strings$(BUILD_EXEEXT)
 	$(AM_V_CCLD)$(CC_FOR_BUILD) $(extract_strings_cppflags) $(CPPFLAGS_FOR_BUILD) $(extract_strings_cflags) $(CFLAGS_FOR_BUILD) $^ $(LDFLAGS_FOR_BUILD) $(extract_strings_ldadd) $(LIBS_FOR_BUILD) -o $@
 
-noinst_PROGRAMS = extract-strings
+all-local: extract-strings$(BUILD_EXEEXT)
 
 -include $(top_srcdir)/git.mk
-- 
1.8.1.2

